install-deps.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -e
+#!/bin/sh -e
```

Comment:
On 22-3-2016 07:41, Mykola Golub wrote:
> In install-deps.sh
> <https://github.com/ceph/ceph/pull/7515#discussion_r56942955>:
> 
>> @@ -1,4 +1,4 @@
>> -#!/bin/bash -e
>> +#!/bin/sh -e
> 
> I vote for adding bash to FreeBSD install-deps.sh and changing all
> |#!/bin/bash| to |#!/usr/bin/env bash| in our scripts.
> 
> It looks like there is not a way to pass bash options when invoking via
> /usr/bin/env. Thus lines like this:
> 
> |#!/bin/bash -e |
> 
> have to be changed to
> 
> |#!/usr/bin/env bash 
> set -e |

I'm sort of at limbo here...
It is definitely the most portable solution, and thus deserves to be chosen.

But:
- It requires changing a lot of scripts.
- And it requires a lot of careful monitoring, because on average basis
people are used to writing scripts by just starting '#!/bin/bash'. And
then we need to go back, and keep changing this over and over, because
it fails on FreeBSD.

Perhaps we should ask others (Sage??) what his preference is?

Either way, I'd be willing to do the initial patching.
And I'd be running into scripts that do not "behave", once I kill my
/bin/bash link.